domains/transfer/TransferAgreementAutomation/Worker.UnitTests/TransferAgreementsAutomationWorkerTests.cs

```diff
@@ -80,25 +88,25 @@ public async Task WhenCalled_CacheIsSetToHealthy()
         using var mockHttpMessageHandler = new MockHttpMessageHandler();
         var memoryCache = new AutomationCache();
 
-        var agreements = new List<TransferAgreementDto>
-        {
-            new(
-                EndDate: DateTimeOffset.Now.AddHours(3).ToUnixTimeSeconds(),
-                ReceiverReference: Guid.NewGuid().ToString(),
-                ReceiverTin: "12345678",
-                SenderId: Guid.NewGuid().ToString(),
-                StartDate: DateTimeOffset.Now.ToUnixTimeSeconds()
-            )
-        };
+        memoryCache.Cache.Get(HealthEntries.Key).Should().BeNull("Cache should initially be empty");
 
-        mockHttpMessageHandler.Expect("/api/all-transfer-agreements").Respond("application/json",
+        var agreements = new List<TransferAgreementDto>
+    {
+        new(
+            EndDate: DateTimeOffset.UtcNow.AddHours(3).ToUnixTimeSeconds(),
+            ReceiverReference: Guid.NewGuid().ToString(),
+            ReceiverTin: "12345678",
+            SenderId: Guid.NewGuid().ToString(),
+            StartDate: DateTimeOffset.UtcNow.ToUnixTimeSeconds()
+        )
+    };
+
+        mockHttpMessageHandler.Expect("/api/internal-transfer-agreements/all").Respond("application/json",
             JsonSerializer.Serialize(new TransferAgreementsDto(agreements)));
 
-        using var cts = new CancellationTokenSource();
-
         poWalletServiceMock
             .When(x => x.TransferCertificates(Arg.Any<TransferAgreementDto>()))
-            .Do(_ => { cts.Cancel(); });
+            .Do(_ => { /* Simulate transfer without cancelling the token */ });
```

Comment:
Why remove this? Wont et resolve in going on forever?

Reply:
The call to TransferCertificates resulted in cancellation of the entire operation.

This approach is generally more aligned with typical usage patterns, where a service method like TransferCertificates would do its job and return normally, without externally influencing the cancellation state of the entire process.

Reply:
It is intented to cancle the operation as soon as possible. We only test the memorycache here.

Reply:
Changed back to original implementation, but with correct Uri, and without using in cancellationtoken: b923fa3